### PR TITLE
extensions for 2D convex shape

### PR DESCRIPTION
**Pull request by MichaelRiss**
Hello Sébastien,

I tried to play around with 2D convex shapes and noticed that they are not completely supported yet. Then I tried to patch in what's missing and to my surprise succeeded in getting a basic 2D convex demo up and running. Maybe you want to take a look at it, comment and - if everything is ok - merge?
This pull request contains the parts of ncollide.

Michi

### CODE REVIEW

File: ncollide_utils/triangle.rs [hunk 1]
```diff
@@ -24,6 +24,14 @@ pub fn triangle_area<P>(pa: &P, pb: &P, pc: &P) -> <P::Vect as Vect>::Scalar
     sqr.sqrt() * na::cast(0.25)
 }
 
+/// Computes the center of a triangle.
+#[inline]
+pub fn triangle_center<N, P>(pa: &P, pb: &P, pc: &P) -> P
+    where N: Scalar,
+          P: Axpy<N> + Mul<N, Output = P> + Copy {
+    ::center(&[ *pa, *pb, *pc])
```
Comment (sebcrozet) [inline ncollide_utils/triangle.rs]: There should be a space after `pc`, i.e. `*pc ]`

**Comment by sebcrozet** [comment]
Thanks!

Besides the style-related comment, could you squash the four commits into a single one please?

Btw, if you want to force cargo to use a local version of the libraries (as your did) you can simply  declare path overrides on your `~/.cargo/config` file. For example:

```
paths = ["/path/to/ncollide_entities", "/path/to/ncollide_math" ] 
```

**Comment by MichaelRiss** [comment]
Style is fixed, commits are squashed and thanks for the hint with the paths variable.

I'll have to see whether it also works with github repositories. I wanted to give you an opportunity to compile and test my changes before pulling them into the main repository.

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**